### PR TITLE
use fieldName in calling missingRequiredFields

### LINKED ISSUE #1247: source code which field name is a keyword outputs an invalid code
the file generated with this protocol buffers source code is not compilable because the generated file uses the keyword as an identifier. I'm making Pull Request to fix this issue.

```proto
syntax = "proto2";

message VarTest {
    required uint32 var = 1;
}
```

generated code
```kotlin
// Code generated by Wire protocol buffer compiler, do not edit.
// Source file: test.proto
import com.squareup.wire.FieldEncoding
import com.squareup.wire.Message
import com.squareup.wire.ProtoAdapter
import com.squareup.wire.ProtoReader
import com.squareup.wire.ProtoWriter
import com.squareup.wire.WireField
import com.squareup.wire.internal.missingRequiredFields
import kotlin.Any
import kotlin.AssertionError
import kotlin.Boolean
import kotlin.Deprecated
import kotlin.DeprecationLevel
import kotlin.Int
import kotlin.Nothing
import kotlin.String
import kotlin.jvm.JvmField
import okio.ByteString

class VarTest(
  @field:WireField(
    tag = 1,
    adapter = "com.squareup.wire.ProtoAdapter#UINT32",
    label = WireField.Label.REQUIRED
  )
  val var_: Int,
  unknownFields: ByteString = ByteString.EMPTY
) : Message<VarTest, Nothing>(ADAPTER, unknownFields) {
  @Deprecated(
    message = "Shouldn't be used in Kotlin",
    level = DeprecationLevel.HIDDEN
  )
  override fun newBuilder(): Nothing = throw AssertionError()

  override fun equals(other: Any?): Boolean {
    if (other === this) return true
    if (other !is VarTest) return false
    return unknownFields == other.unknownFields
        && var_ == other.var_
  }

  override fun hashCode(): Int {
    var result = super.hashCode
    if (result == 0) {
      result = unknownFields.hashCode()
      result = result * 37 + var_.hashCode()
      super.hashCode = result
    }
    return result
  }

  override fun toString(): String {
    val result = mutableListOf<String>()
    result += """var_=$var_"""
    return result.joinToString(prefix = "VarTest{", separator = ", ", postfix = "}")
  }

  fun copy(var_: Int = this.var_, unknownFields: ByteString = this.unknownFields): VarTest =
      VarTest(var_, unknownFields)

  companion object {
    @JvmField
    val ADAPTER: ProtoAdapter<VarTest> = object : ProtoAdapter<VarTest>(
      FieldEncoding.LENGTH_DELIMITED, 
      VarTest::class
    ) {
      override fun encodedSize(value: VarTest): Int = 
        ProtoAdapter.UINT32.encodedSizeWithTag(1, value.var_) +
        value.unknownFields.size

      override fun encode(writer: ProtoWriter, value: VarTest) {
        ProtoAdapter.UINT32.encodeWithTag(writer, 1, value.var_)
        writer.writeBytes(value.unknownFields)
      }

      override fun decode(reader: ProtoReader): VarTest {
        var var_: Int? = null
        val unknownFields = reader.forEachTag { tag ->
          when (tag) {
            1 -> var_ = ProtoAdapter.UINT32.decode(reader)
            else -> reader.readUnknownField(tag)
          }
        }
        return VarTest(
          var_ = var_ ?: throw missingRequiredFields(var, "var"),
          unknownFields = unknownFields
        )
      }

      override fun redact(value: VarTest): VarTest = value.copy(
        unknownFields = ByteString.EMPTY
      )
    }
  }
}
```

### PR DESCRIPTION
**Pull request by anatawa12**
fixes #1247.

### CODE REVIEW

**Comment by anatawa12** [comment]
I failed the test for the project ':wire-gradle-plugin'. but I checked that I can build successfully project 'wire-kotlin-generator'. I don't know why I can't pass the test for the project ':wire-gradle-plugin'.

File: wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt [hunk 1]
```diff
@@ -938,7 +938,7 @@ class KotlinGenerator private constructor(
         val fieldName = nameAllocator[field]
 
         val throwExceptionBlock = if (!field.isRepeated && !field.isMap && field.isRequired) {
-          CodeBlock.of(" ?: throw %1M(%2L, %2S)", missingRequiredFields, field.name())
+          CodeBlock.of(" ?: throw %1M(%2L, %3S)", missingRequiredFields, fieldName, field.name())
```
Comment (Egorand) [inline wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: Please add a test for this change.

Comment (anatawa12) [reply wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: I can't make a good test name. Could you rename the test name or could you tell me the good name of the test?

Comment (Egorand) [reply wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: It looks good!

**Comment by Egorand** [comment]
Thanks!

**Pull request merged**